cmd/launcher/desktop.go

```diff
@@ -180,3 +204,24 @@ func defaultSocketPath() string {
 
 	return filepath.Join(os.TempDir(), fmt.Sprintf("%s_%d", socketBaseName, os.Getpid()))
 }
+
+func defaultStatusPath() string {
+	const statusBaseName = "desktop_status.json"
+
+	return filepath.Join(os.TempDir(), fmt.Sprintf("%s_%d", statusBaseName, os.Getpid()))
+}
+
+// Loads menu data from the desktop status file
+func getMenuData(logger log.Logger, statusPath string) *menu.MenuData {
+	statusFileBytes, err := os.ReadFile(statusPath)
+	if err != nil {
+		level.Error(logger).Log("msg", "failed to read desktop status file", "path", statusPath)
+	}
+
+	var desktopStatus runner.DesktopUserStatus
+	if err := json.Unmarshal(statusFileBytes, &desktopStatus); err != nil {
+		level.Error(logger).Log("msg", "failed to unmarshal desktop status json")
+	}
```

Comment:
```suggestion
		level.Error(logger).Log("msg", "failed to unmarshal desktop status json")
		return
	}
```

Reply:
This was moved to `menu.getMenuData()` but fixed it there.